### PR TITLE
[MM-37712] Bubble DragEvent to drag and drop text into input/textarea

### PR DESCRIPTION
**Pull request by sk409**
#### Summary

Bubble DragEvent to drag and drop text into input/textarea.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/18074

#### Related Pull Requests

No related PRs.

##### before

https://user-images.githubusercontent.com/25968819/192328143-3f703dc7-d776-4157-a097-914342a8f299.mov

##### after

https://user-images.githubusercontent.com/25968819/192328184-cf37e2b5-9935-45bf-88eb-4b10dd0ee8e6.mov

#### Release Note

```release-note
Text can be dragged and droppped into input-fields.
```

### CODE REVIEW

**Review by neallred — APPROVED** [review]
LGTM, thanks @sk409 ! Tried the functionality on a variety of inputs in the main app, the system console, and various settings modal, and it all seemed to work correctly without unexpected side-effects. The e2e test run (https://automation-dashboard.vercel.app/cycles/4c255ab1-fd12-4889-89b4-6d99c48c000d) also looks good to me (comparing to a recent run against master branch).

This won't enable drag and drop of text for the invite to team/invite to channel modals, but I wouldn't expect it to since the HTML backing them isn't plain inputs/textareas. Had a brief look at trying to do that, and I don't think it would be worth trying in this PR.

File: utils/dragster.ts [hunk 1]
```diff
@@ -9,20 +9,22 @@ type Options = {
 }
 
 export default function dragster(query: string, options: Options) {
-    const noop = () => {}; // eslint-disable-line no-empty-function
+    const preventDefault = (e: Event) => {
```
Comment (hmhealey) [inline utils/dragster.ts:12]: Are these changes to these default handlers needed when `FileUpload` overrides all of these anyway?

If I'm reading this correctly, it means that all unhandled drag events will be blocked as soon as someone else starts using dragster

**Review by hmhealey — CHANGES_REQUESTED** [review]
Thanks for the PR! Glad to get this sorted out, but I did have one question about this before I can approve it

File: utils/dragster.ts (hunk 1 above)
Comment (sk409) [reply utils/dragster.ts:12]: @hmhealey 
Thank you for your review!

> If I'm reading this correctly, it means that all unhandled drag events will be blocked as soon as someone else starts using dragster

Yes, you are correct.
But I didn't change behavior of dragster.
dragenter, dragleave, dragover and drop called preventDefault before I made this changes.
But I needed to lifted preventDefault up to FileUpload so I replaced noop with preventDefault to keep behavior compatible.

**Comment by jgilliam17** [comment]
Thanks @sk409 apologies for delay in getting to this PR; can you please update the branch before we merge?
Tested, looks good to merge.
Verified drag and drop into text input is working as expected. Thanks for running e2es @neallred

**Review by jgilliam17 — APPROVED** [review]

**Pull request merged**